src/app/components/OrderInput.tsx

```diff
@@ -0,0 +1,435 @@
+        {pairToken1.symbol}
+      </button>
+      <button
+        className={
+          "btn" +
+          (selecedToken.address === pairToken2.address ? " btn-active" : "")
+        }
+        onClick={() => {
+          dispatch(orderInputSlice.actions.setToken1Selected(false));
+        }}
+      >
+        {pairToken2.symbol}
+      </button>
+    </div>
+  );
+}
+
+function PositionSizeInput() {
+  const defaultValue = useAppSelector((state) => state.orderInput.size);
+  const selectedToken = useAppSelector(getSelectedToken);
+  const validationResult = useAppSelector(validatePositionSize);
+  const dispatch = useAppDispatch();
+  return (
+    <div className="form-control">
+      <label className="label">
+        <span className="label-text">Position Size</span>
+      </label>
+      <div className="relative">
+        <input
+          type="number"
+          className={
+            "input input-bordered w-full" +
+            (validationResult.valid ? "" : " input-error")
+          }
+          value={defaultValue}
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            const size = Number(event.target.value);
+            dispatch(orderInputSlice.actions.setSize(size));
+          }}
+        ></input>
+
+        <div className="absolute right-2 top-1/2 -translate-y-1/2 flex items-center space-x-2">
+          <img src={selectedToken.iconUrl} className="w-6 h-6 my-auto mx-1" />
+          <span>{selectedToken.symbol}</span>
+        </div>
+      </div>
+      <label className="label">
+        <span className="label-text-alt text-error">
+          {validationResult.valid ? "" : validationResult.message}
+        </span>
+      </label>
+
+      <div className="btn-group w-full">
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(25))}
+        >
+          25%
+        </button>
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(50))}
+        >
+          50%
+        </button>
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(75))}
+        >
+          75%
+        </button>
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(100))}
+        >
+          100%
+        </button>
+        <input
+          type="number"
+          className="input input-sm bg-neutral"
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            const size = Number(event.target.value);
+            dispatch(setSizePercent(size));
+          }}
+        ></input>
+      </div>
+    </div>
+  );
+}
+
+// TODO: test if floating point numbers are handled correctly
+function slippagePercentage(slippage: number): string {
+  return displayNumber(slippage * 100, 0);
+}
+
+function slippageFromPercentage(percentage: string): number {
+  // TODO: decimal numbers with dots (1.3) don't work (but 1,3 does)
```

Comment:
I find the opposite, 1.3 does work but 1,3 doesn't (the input box doesn't allow a ,

Reply:
maybe something to do with locale... created #76